src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,104 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a TaskList.
+ */
+public class TaskList {
+    private ArrayList<Task> taskList;
+
+    /**
+     * Constructs a TaskList object.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<Task>();
+    }
+
+    /**
+     * Constructs a TaskList object.
+     * 
+     * @param savedTaskList ArrayList of type Task retrieved from saved data.
+     * @throws DukeException If there is an error loading the file.
+     */
+    public TaskList(ArrayList<Task> savedTaskList) throws DukeException {
+        // this.taskList.addAll(savedTaskList);
+        this.taskList = savedTaskList;
+    }
+
+    /**
+     * Returns the size of the task list.
+     * @return Size of the task list as an integer.
+     */
+    public int size() {
+        return taskList.size();
+    }
+
+    /**
+     * Adds a task to the task list.
+     * @param task Task object to be added to the task list.
+     */
+    public void addToTaskList(Task task) {
+        if (taskList.size() < 100) {
+            taskList.add(task);
+        } else {
+            System.out.println("Error: List is full.");
+        }
+    }
+
+    /**
+     * Returns the task at the specified index.
+     * @param i Index of the task of interest.
+     * @return String representation of the task at the specified index with numbering.
+     */
+    public String taskToString(int i) {
```

Comment:
nit: perhaps could change the name to prevent confusion with the toString method!

Reply:
Got it. Will do!